src/main/java/de/tum/in/www1/artemis/web/rest/CompetencyResource.java

```diff
@@ -576,4 +621,21 @@ private void checkAuthorizationForCompetency(Role role, @NotNull Course course,
         }
         authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(role, course, null);
     }
+
+    /**
+     * Creates a new Competency from an existing one to save to the database
+     *
+     * @param competency the existing competency
+     * @return the new Competency
+     */
+    public Competency getCompetencyToCreate(Competency competency) {
+        Competency competencyToCreate = new Competency();
+        competencyToCreate.setTitle(competency.getTitle().trim());
+        competencyToCreate.setDescription(competency.getDescription());
+        competencyToCreate.setSoftDueDate(competency.getSoftDueDate());
+        competencyToCreate.setTaxonomy(competency.getTaxonomy());
+        competencyToCreate.setMasteryThreshold(competency.getMasteryThreshold());
+        competencyToCreate.setOptional(competency.isOptional());
+        return competencyToCreate;
+    }
```

Comment:
Can you add a Copy Constructor to the `Competency` class instead of this method?

Reply:
Solving this only with a copy constructor is not a good idea imo, since we want to omit some attributes (exercises, lectureUnits, ...).
I added a full copy constructor and kept the method (which now simply calls the copy constructor and removes the unwanted attributes.